web/packages/teleport/src/Main/Main.tsx

```diff
@@ -169,13 +169,47 @@ export function Main(props: MainProps) {
   );
 }
 
-function renderRoutes(features: TeleportFeature[]) {
+function renderRoutes(
+  features: TeleportFeature[],
+  lockedFeatures: LockedFeatures
+) {
   const routes = [];
 
+  const lockedParents = getLockedParents(features, lockedFeatures);
+
   for (const [index, feature] of features.entries()) {
+    const isParentLocked = lockedParents.find(
+      parent => feature.parent && parent === feature.parent.name
+    );
```

Comment:
This works and looks way better. It does instantiate `feature.parent` multiple times unnecessarily though, since all children of a feature will instantiate it and run `isLocked`. 

I don't have a strong opinion, let me know what you think.

Reply:
Yeah, this looks way better. Thank you.